activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -526,12 +526,15 @@ def build_insert_sql(insert) # :nodoc:
         sql
       end
 
-      private
-        def check_version
-          if version < "5.5.8"
-            raise "Your version of MySQL (#{version_string}) is too old. Active Record supports MySQL >= 5.5.8."
-          end
+      def check_version!
+        if version < "5.5.8"
+          raise "Your version of MySQL (#{version_string}) is too old. Active Record supports MySQL >= 5.5.8."
+        else
+          schema_cache.database_version_checked = true
```

Comment:
This goes in the opposite direction of what we should be doing. For every other datapoint the schema cache manages the caching and calls out to connection if needed.